src/backend/distributed/transaction/transaction_management.c

```diff
@@ -262,6 +266,11 @@ CoordinatedTransactionCallback(XactEvent event, void *arg)
 				AfterXactConnectionHandling(true);
 			}
 
+			if (MetadataSyncOnCommit)
+			{
+				TriggerMetadataSync(MyDatabaseId);
```

Comment:
I guess this is still a bit of a race condition, but the window is short and the cost of the metadata sync still starting too early, so probably ok.

One thing we could consider is to put this in post-commit, because at that point the record is actually visible.